### PR TITLE
skip tests if requirements for spectrum functions are not met

### LINKED ISSUE #326: Tests that require skypy-data are not skipped if it is not installed
**Describe the bug**
The tests [`test_load_spectral_data`](https://github.com/skypyproject/skypy/blob/master/skypy/galaxy/tests/test_spectrum.py) and [`test_spectral_data_input`](https://github.com/skypyproject/skypy/blob/master/skypy/utils/tests/test_decorators.py) require [`skypy-data`](https://github.com/skypyproject/skypy-data) but are not skipped if it is not installed leading to test failures.

**Expected behaviour**
These tests should be skipped if `skypy-data` is not installed. As an example see how the `HAS_SPECUITLS` flag is used in [`test_spectrum.py`](https://github.com/skypyproject/skypy/blob/master/skypy/galaxy/tests/test_spectrum.py) to similarly skip some tests when `specutils` is no installed.

**To Reproduce**
```
pip install skypy==0.3rc2 pytest-astropy pytest-rerunfailures specutils
pytest --pyargs skypy.galaxy.tests.test_spectrum skypy.utils.tests.test_decorators
```

### PR DESCRIPTION
**Pull request by ntessore**
## Description

Skip tests that require `skypy-data` if the package is not available. Closes #326.

Now also does the right thing in each case when specutils is not installed.

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

File: skypy/utils/tests/test_decorators.py [hunk 1]
```diff
@@ -82,6 +83,8 @@ def argument_z_does_not_exist(x, y):
             pass
 
 
+@pytest.mark.skipif(not HAS_SPECUTILS or not HAS_SKYPY_DATA,
```
Comment (philipp128) [inline skypy/utils/tests/test_decorators.py:86]: Does this test really need `specutils`?

Comment (ntessore) [reply skypy/utils/tests/test_decorators.py:86]: Yes

**Review by philipp128 — APPROVED** [review]
Looks good to me.

File: skypy/galaxy/_spectrum_loaders.py [hunk 2]
```diff
@@ -11,6 +10,11 @@
 import urllib
 from pkg_resources import resource_filename
 
+try:
```
Comment (ntessore) [inline skypy/galaxy/_spectrum_loaders.py:13]:
```suggestion
# this file is only ever imported when specutils is present
# but without the try/except pytest will fail when doctests are discovered
try:
```

**Review by rrjbca — APPROVED** [review]

**Pull request merged**